src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/HPackDecoder.cs

```diff
@@ -397,14 +545,14 @@ public void CompleteDecode()
 
         private void OnIndexedHeaderField(int index, IHttpHeadersHandler? handler)
         {
-            HeaderField header = GetHeader(index);
+            ref readonly HeaderField header = ref GetHeader(index);
             handler?.OnHeader(header.Name, header.Value);
             _state = State.Ready;
         }
 
         private void OnIndexedHeaderName(int index)
         {
-            HeaderField header = GetHeader(index);
+            ref readonly HeaderField header = ref GetHeader(index);
             _headerName = header.Name;
             _headerNameLength = header.Name.Length;
             _state = State.HeaderValueLength;
```

Comment:
I expect this would be ever-so-slightly better as:
```C#
byte[] name = GetHeader(index).Name;
_headerName = name;
_headerNameLength = name.Length;
_state = State.HeaderValueLength;
```